src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -193,16 +196,178 @@ public override SyntaxToken VisitToken(SyntaxToken token)
                 }
 
                 // the next token gets the accrued trivia.
-                if (result.Kind() != SyntaxKind.None && _residualTrivia != null && _residualTrivia.Count > 0)
+                if (result.Kind() != SyntaxKind.None && HasResidualTrivia)
                 {
                     _residualTrivia.Add(result.LeadingTrivia);
-                    result = result.WithLeadingTrivia(_residualTrivia.ToList());
+                    result = result.WithLeadingTrivia(ProcessElasticMarkers(_residualTrivia.ToList()));
                     _residualTrivia.Clear();
                 }
 
                 return result;
             }
 
+            public override SyntaxTriviaList VisitList(SyntaxTriviaList list)
```

Comment:
📝 This is mostly derived from the base implementation, with two notable changes:

1. `ResidualTrivia` is cleared before visiting structured trivia. Otherwise, `VisitToken` for a token nested within structured trivia could incorrectly relocate the residual trivia relative to its intended location (at the beginning of the trivia list of the parent token of the structured trivia).
2. `ResidualTrivia` is checked after visiting each trivia. If a node was removed inside of structured trivia, it may produce residual trivia that needs to be inserted within the containing trivia list and not relocated to the beginning of the list.

Aside from the call to `ProcessElasticMarkers`, this method is strictly bug fixes relative to the original implementation.